make sidebar collapsible

TiziG opened the pull request:
remove unnecessary overflow:visible (#sidebar_plugin_SpoolManager) css rule because it overrides the visibility setting in the .collapse class.

This fixes issue #197

Alokin-ITA commented:
This will make the `dropdown-menu` being cut at the bottom as a result of it. In order to avoid this, the button `dropdown-toggle` and the `dropdown-menu` can be moved on the right side of the `accordion-heading`, just like the "Files" widget does.
When the widget is collapsed the `dropdown-toggle` button should disappear.
With `dropdown-menu` out of the collapsible `accordion-body`, it will not suffer from `overflow:hidden` and this commit is acceptable.

TiziG commented:
I replaced the dropdown with a modal dialog.

Please let me know if I missed something.

## Changes
- Removed the dropdown buttons
- Created a modal dialog for selecting spools
- Added change filament buttons
- Moved the filter checkboxes into the new dialog

TiziG commented:
Also fixed #190

OllisGit commented:
@TiziG 
> Let me know if I can help with something.

Well...maybe you can update your other pull request #206 , so that there is no merge conflict ay more ;-)
